src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTemplateEngineFactoryService.cs

```diff
@@ -70,8 +73,36 @@ public override RazorTemplateEngine Create(string projectPath, Action<IRazorEngi
 
         private Version GetMvcVersion(string projectPath)
         {
-            // TODO: Need to figure out the actual referenced Mvc version in the project. https://github.com/aspnet/Razor/issues/1183
+            var workspace = _services.WorkspaceServices.Workspace;
+
+            var project = workspace.CurrentSolution.Projects.FirstOrDefault(p =>
+            {
+                var directory = Path.GetDirectoryName(p.FilePath);
+                return string.Equals(
+                    NormalizeDirectoryPath(directory),
+                    NormalizeDirectoryPath(projectPath),
+                    StringComparison.OrdinalIgnoreCase);
+            });
+
+            if (project != null)
+            {
+                var compilation = project.GetCompilationAsync().Result;
```

Comment:
I still need to compilation to get the `IAssemblySymbol` from the `MetadataReference`.